packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -28,6 +28,52 @@ const int _kDefaultSizeBytes = 100 << 20; // 100 MiB
 ///
 /// A shared instance of this cache is retained by [PaintingBinding] and can be
 /// obtained via the [imageCache] top-level property in the [painting] library.
+///
+/// {@tool sample}
+///
+/// This sample shows how to supply your own caching logic and replace the
+/// global [imageCache] varible.
+///
+/// ```dart
+/// import 'package:flutter/material.dart';
+/// import 'package:flutter/foundation.dart' as foundation;
+/// import 'package:flutter/gestures.dart' as gestures;
+/// import 'package:flutter/rendering.dart' as rendering;
+/// import 'package:flutter/scheduler.dart' as scheduler;
+/// import 'package:flutter/services.dart' as services;
+/// import 'package:flutter/widgets.dart' as widgets;
+///
+/// class MyImageCache extends ImageCache {}
+///
+/// class MyWidgetsBinding extends foundation.BindingBase
+///     with
+///         gestures.GestureBinding,
+///         services.ServicesBinding,
+///         scheduler.SchedulerBinding,
+///         rendering.PaintingBinding,
+///         rendering.SemanticsBinding,
+///         rendering.RendererBinding,
+///         widgets.WidgetsBinding {
+///
+///   @override
+///   widgets.ImageCache createImageCache() => MyImageCache();
```

Comment:
We should include the `MyImageCache` implementation somewhere in this sample - I believe analysis will fail without it anyway.  It could be something very simple, just extend ImageCache and override one member or something.

Reply:
It was in there, I added an example override to make it more visible.